src/Mod/Path/PathScripts/features/PathFeatureExtensions.py

```diff
@@ -0,0 +1,305 @@
+
+        return self._getDirectedNormal(e0.valueAt(midparam), normal.normalize())
+
+    def getWire(self):
+        PathLog.track()
+        if PathGeom.isRoughly(0, self.length.Value) or not self.sub:
+            PathLog.debug("no extension, length=%.2f, sub=%s" % (self.length.Value, self.sub))
+            return None
+
+        feature = self.obj.Shape.getElement(self.feature)
+        edges = self._getEdges()
+        sub = Part.Wire(Part.sortEdges(edges)[0])
+
+        if 1 == len(edges):
+            PathLog.debug("Extending single edge wire")
+            edge = edges[0]
+            if Part.Circle == type(edge.Curve):
+                circle = edge.Curve
+                # for a circle we have to figure out if it's a hole or a cylinder
+                p0 = edge.valueAt(edge.FirstParameter)
+                normal = (edge.Curve.Center - p0).normalize()
+                direction = self._getDirectedNormal(p0, normal)
+                if direction is None:
+                    return None
+
+                if PathGeom.pointsCoincide(normal, direction):
+                    r = circle.Radius - self.length.Value
+                else:
+                    r = circle.Radius + self.length.Value
+
+                # assuming the offset produces a valid circle - go for it
+                if r > 0:
+                    e3 = Part.makeCircle(r, circle.Center, circle.Axis, edge.FirstParameter * 180 / math.pi, edge.LastParameter * 180 / math.pi)
+                    if endPoints(edge):
+                        # need to construct the arc slice
+                        e0 = Part.makeLine(edge.valueAt(edge.FirstParameter), e3.valueAt(e3.FirstParameter))
+                        e2 = Part.makeLine(edge.valueAt(edge.LastParameter), e3.valueAt(e3.LastParameter))
+                        return Part.Wire([e0, edge, e2, e3])
+
+                    return Part.Wire([e3])
+
+                # the extension is bigger than the hole - so let's just cover the whole hole
+                if endPoints(edge):
+                    # if the resulting arc is smaller than the radius, create a pie slice
+                    PathLog.track()
+                    center = circle.Center
+                    e0 = Part.makeLine(center, edge.valueAt(edge.FirstParameter))
+                    e2 = Part.makeLine(edge.valueAt(edge.LastParameter), center)
+                    return Part.Wire([e0, edge, e2])
+
+                PathLog.track()
+                return Part.Wire([edge])
+
+            else:
+                PathLog.track(self.feature, self.sub, type(edge.Curve), endPoints(edge))
+                direction = self._getDirection(sub)
+                if direction is None:
+                    return None
+
+            #    return self._extendEdge(feature, edge, direction)
+            return self._extendEdge(feature, edges[0], direction)
+
+        return extendWire(feature, sub, self.length.Value)
+# Eclass
+
+
+def initialize_properties(obj):
```

Comment:
Please don't use underscore naming convention, everything else is in camel case so let's stick with that.

Reply:
Done.